hardware-api.md

```diff
@@ -91,7 +97,23 @@ pin.analogRead(function(error, value) {
 
 ### PWM pins
 
-PWM pins are not yet implemented. See [#21](https://github.com/tessel/t2-firmware/issues/21).
+**PWM pins** are pulse-width modulated pins ([wiki link](http://en.wikipedia.org/wiki/Pulse-width_modulation)). Essentially, PWM is a digital signal that spends between 0% and 100% of its time on (this is its "duty cycle"). You can set the PWM pins to any value between 0 and 1 to approximate an analog signal. PWM is often used to control servo speeds or LED brightness.
```

Comment:
:+1: to the first comment. 

I don't think this API documentation is the place to explain the details of electrical theory. My hope is that if they are interested, they click in the link already provided.